Security/FeatureSecurity.php

```diff
@@ -32,6 +32,12 @@ public function __construct(AuthorizationCheckerInterface $context = null)
      */
     public function isGranted(Feature $feature)
     {
+        // feature is enabled without required roles
+        // there's no need to check on user roles
+        if (empty($feature->getRole()) && empty($feature->getParentRole())) {
```

Comment:
I think that you can delegate this two check to the feature entity: something like `hasRolesAttached`that checks on the feature itself and on the parent (the code is a draft, please double check):
```
class Feature
{
    public function hasRolesAttached()
    {
        return $this->role || $this->parentHasRolesAttached();
    }

    private function parentHasRolesAttached()
    {
        return $this->parent && $this->parent->hasRolesAttached();
    }
}
```